ktor-utils/jvm/src/io/ktor/util/Cache.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util
+
+import java.util.*
+
+/**
+ * Initial capacity of [LinkedHashMap] used as [LRUCache].
+ */
+private const val CACHE_INITIAL_CAPACITY = 10
+
+/**
+ * Load factory of [LinkedHashMap] used as [LRUCache].
+ */
+private const val CACHE_LOAD_FACTOR = 0.75f
+
+/**
+ * Create a new instance of thread safe [LRUCache] and return it.
+ */
+@InternalAPI
+fun <K, V> createLRUCache(supplier: (K) -> V, close: (V) -> Unit, maxSize: Int): Map<K, V> =
+    Collections.synchronizedMap(LRUCache(supplier, close, maxSize))
+
+/**
+ * LRU cache based on [LinkedHashMap] with specified [maxSize] and [supplier].
+ */
+@InternalAPI
+class LRUCache<K, V> internal constructor(
```

Comment:
I believe it could be a private or internal class: no need to track it's binary compatibility since we don't create it directly without `createLRUCache` function.

Reply:
Good point, done.